pkg/packages.go

```diff
@@ -58,55 +58,61 @@ func Install(ctx context.Context, dependencySourceIdentifier string, m spec.Json
 		if err != nil {
 			return nil, errors.Wrap(err, "failed to install package")
 		}
-		lock.Dependencies, err = insertDependency(lock.Dependencies, spec.Dependency{
-			Name:      dep.Name,
-			Source:    dep.Source,
-			Version:   lockVersion,
-			DepSource: dependencySourceIdentifier,
-		})
-		if err != nil {
-			return nil, errors.Wrap(err, "failed to insert dependency to lock dependencies")
-		}
 
-		destPath := path.Join(dir, dep.Name)
-		if err != nil {
-			return nil, errors.Wrap(err, "failed to find destination path for package")
-		}
+		// If dependencies are being installed from a lock file, the transitive
+		// dependencies are not questioned, but the locked dependencies are
+		// just installed.
+		if !isLock {
```

Comment:
very small nit: is `if isLock { continue }` and unindenting the whole block below more readable/understandable?

Reply:
agreed